### DISCUSSION TITLE
How to change the default "Project Location"

### DISCUSSION THREAD

**Discussion by ghost** [apache/netbeans#6485] [Q&A]
System info:
```
Operating System: Manjaro Linux 
KDE Plasma Version: 5.27.8
KDE Frameworks Version: 5.110.0
Qt Version: 5.15.10
Kernel Version: 6.5.5-1-MANJARO (64-bit)
Graphics Platform: X11
Graphics Processor: Mesa Intel® Graphics
```

When i create a new project:

![Screenshot_20230925_030903](https://github.com/apache/netbeans/assets/124194790/65505d84-0b53-4884-9f0f-52d5c6bc3f72)

I wanna change the default project location so that it it hardcoded and set automatically to my custom location when creating subsequent projects.

The default project location is: `/home/user0/NetBeansProjects/`

I wanna change it to: `/home/user0/Projects/java/`

Netbeans is manually installed in: `/home/user0/Programs/netbeans/`

Where is the configuration file where that setting of "Project Location" is set ? 

I tried searching in `/home/user0/.netbeans` but there are so many results and it's tedious, also i tried searching in the settings from within Netbeans and i found nothing there.

**Comment by mbien** [comment]
The new-project wizard should remember the last entered location and automatically set it next time to create a project.

**Reply by ghost** [reply]
Yes i know it should be set automatically, bu it's not for some reason. I think i have to set it manually from some config file.

**Reply by manueljordan** [reply]
I confirm the same behavior for Netbeans 31 too. For a permanent edition is mandatory edit the `~/.netbeans/31/config/Preferences/org/netbeans/modules$ nano -v projectui.properties` file

**Comment by p-i-c-o** [comment]
In the file: `/home/<your_user>/.netbeans/<your_number>/config/Preferences/org/netbeans/modules/projectui.properties`
add:
```
lastOpenProjectDir=/path/to/prefered/dir
```

> tested on Linux Mint 22.1 x86_64 using NetBeans v25 from Flatpak

**Reply by mbien** [reply]
you can also set the initial default with `-J-Dnetbeans.projects.dir=/path/to/projects`
